kong/router/compat.lua

```diff
@@ -235,9 +227,11 @@ local function get_priority(route)
     match_weight = match_weight + 1
   end
 
-  local plain_host_only = not not hosts
+  local plain_host_only = not is_empty_field(hosts)
```

Comment:
This change might not be correct, because it treats `hosts = {}` as not having `plain_host_only`. This contradicts with the traditional logic:

https://github.com/Kong/kong/blob/master/kong/router/traditional.lua#L350-L352

which considers `hosts = {}` as `plain_host_only = true`, but `hosts = nil` as `plain_host_only = false`.

Reply:
OK, I will check and fix it.

Reply:
Now the code change to `type(hosts) == "table"`,
I think it is good enough to fit `hosts = {}/nil/ngx.null`.